crates/steel-core/src/compiler/code_gen.rs

```diff
@@ -907,6 +907,17 @@ impl<'a> VisitorMut for CodeGenerator<'a> {
 
         Ok(())
     }
+
+    fn visit_vector(&mut self, v: &crate::parser::ast::Vector) -> Self::Output {
+        for arg in &v.args {
+            self.visit(arg)?;
+        }
+
+        let payload_size = 2 * v.args.len() + usize::from(v.bytes);
```

Comment:
The payload for `VEC` is a mixture of length and byte-ness, with a bit of bit-fiddling.